server.go

```diff
@@ -1248,16 +1210,6 @@ func (s *server) handleAddPeerMsg(state *peerState, sp *serverPeer) bool {
 
 	// TODO: Check for max peers from a single IP.
 
-	// Limit max outbound peers.
-	if _, ok := state.pendingPeers[sp.Addr()]; ok {
-		if state.OutboundCount() >= state.maxOutboundPeers {
-			srvrLog.Infof("Max outbound peers reached [%d] - disconnecting "+
-				"peer %s", state.maxOutboundPeers, sp)
-			sp.Disconnect()
-			return false
-		}
-	}
-
 	// Limit max number of total peers.
 	if state.Count() >= cfg.MaxPeers {
```

Comment:
As @tuxcanfly pointed out, the connection manager only handles outbound connections, so this code does in fact have to be in the server to limit overall connections.  My previous comment can be ignored.